File: Example/VISPER/Example3/Controller/Example3VisperViewController.m
//
//  Example3VisperViewController.m
//  VISPER
//
//  Created by Bartel on 12.07.15.
//  Copyright (c) 2015 Jan Bartel. All rights reserved.
//

#import "Example3VisperViewController.h"
#import <VISPER/UIViewController+VISPER.h>
@interface Example3VisperViewController ()

@end

@implementation Example3VisperViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"Example3VisperVC";
    // Do any additional setup after loading the view from its nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)closeViewController:(id)sender {
    NSObject<IVISPERViewEvent>*event = [self.visperServiceProvider createEventWithName:@"shouldCloseViewController"
                                                                                sender:self
                                                                                  info:@{}];
    [self notifyPresentersOfView:self.view withEvent:event];
}
@end

File: Example/VISPER/Example3/Controller/Example3VisperViewController.swift
//
//  Example3VisperViewController.swift
//  VISPER
//
//  Created by Jan Bartel on 09.03.16.
//  Copyright © 2016 Jan Bartel. All rights reserved.
//

import Foundation
import VISPER

class Example3VisperViewController: UIViewController {

    
    @IBOutlet weak var textLabel: UILabel!
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Example3VisperVC";
    }
    
    
    @IBAction func closeViewController(sender: AnyObject) {
        
        let event = self.visperServiceProvider.createEventWithName("shouldCloseViewController", sender: self, info: [NSObject : AnyObject]())
       
        self.notifyPresentersOfView(self.view, withEvent: event)
        
    }
    
    @IBAction func loadDataWithSwift(sender: AnyObject) {
        let event = self.visperServiceProvider.createEventWithName("loadDataWithSwift", sender: self, info: [NSObject : AnyObject]())
        
        self.notifyPresentersOfView(self.view, withEvent: event)
    }
    
    func setText(text:String){
        self.textLabel.text = text
    }
}
